File: 2017/11/solution.js
const fs = require('fs')
let inp = fs.readFileSync("./day11input").toString('utf-8').trim().split(","),
    dirs = {'n': [-1,1,0], 'ne': [0,1,-1], 'se': [1,0,-1], 's': [1,-1,0], 'sw': [0,-1,1], 'nw': [-1,0,1]},
    coords = [0,0,0],
    max = -Infinity,
    distance = (x => x.map(Math.abs).reduce((a,b) => a > b ? a : b))

for (let d of inp) {
  coords = coords.map( (x,i) => x + dirs[d][i] )
  max = Math.max(max, distance(coords))
}
console.log(distance(coords), max);

File: 2017/11/solution.py
f = open("11.txt", "r")
ds = f.read().split(",")

x = 0
y = 0
z = 0

dists = []

for d in ds:
  if d == "n":
    y += 1
    z -= 1
  elif d == "s":
    y -= 1
    z += 1
  elif d == "ne":
    x += 1
    z -= 1
  elif d == "sw":
    x -= 1
    z += 1
  elif d == "nw":
    x -= 1
    y += 1
  elif d == "se":
    x += 1
    y -= 1
  dists.append((abs(x) + abs(y) + abs(z)) / 2)

print (abs(x) + abs(y) + abs(z)) / 2
print max(dists)
